spec/Brille24/SyliusCustomerOptionsPlugin/Importer/CustomerOptionPriceCsvImporterSpec.php

```diff
@@ -0,0 +1,154 @@
+            Argument::type('int'),
+            Argument::type('float')
+        )->shouldBeCalledTimes(1)->willReturn($valuePrice);
+
+        $entityManager->persist(Argument::type(CustomerOptionValuePriceInterface::class))->shouldBeCalledTimes(3);
+        $entityManager->flush()->shouldBeCalled();
+
+        $this->import($this->getValidCsvFile());
+    }
+
+    public function it_sends_mail_on_failed_import(
+        CustomerOptionPriceUpdaterInterface $priceUpdater,
+        EntityManagerInterface $entityManager,
+        SenderInterface $sender,
+        TokenStorageInterface $tokenStorage,
+        TokenInterface $token,
+        AdminUserInterface $adminUser,
+        ProductRepositoryInterface $productRepository
+    ): void {
+        $tokenStorage->getToken()->willReturn($token);
+        $token->getUser()->willReturn($adminUser);
+        $adminUser->getEmail()->willReturn('john.doe@example.com');
+
+        $productRepository->findOneByCode(Argument::type('string'))->willThrow(\InvalidArgumentException::class);
+
+        $priceUpdater->updateForProduct(
+            Argument::type('string'),
+            Argument::type('string'),
+            Argument::type('string'),
+            Argument::type(ProductInterface::class),
+            Argument::type('string'),
+            Argument::type('string'),
+            Argument::type('string'),
+            Argument::type('int'),
+            Argument::type('float')
+        )->shouldNotBeCalled();
+
+        $priceUpdater->updateForProduct(
+            Argument::type('string'),
+            Argument::type('string'),
+            Argument::type('string'),
+            Argument::type(ProductInterface::class),
+            null,
+            null,
+            Argument::type('string'),
+            Argument::type('int'),
+            Argument::type('float')
+        )->shouldNotBeCalled();
+
+        $sender->send(Argument::type('string'), ['john.doe@example.com'], Argument::type('array'), Argument::type('array'))->shouldBeCalled();
+
+        $entityManager->persist(Argument::type(CustomerOptionValuePriceInterface::class))->shouldNotBeCalled();
+        $entityManager->flush()->shouldBeCalled();
+
+        $this->import($this->getInValidCsvFile());
+    }
+
+    /**
+     * @return string
+     */
+    private function getValidCsvFile(): string
+    {
+        $csvString = <<<EOT
+product_code,customer_option_code,customer_option_value_code,channel_code,type,amount,percent,valid_from,valid_to
+first_product,lens_type,none,glasses24,FIXED_AMOUNT,1000,0.0,,
+first_product,lens_type,none,glasses24,FIXED_AMOUNT,1500,0.0,2020-01-01,2020-02-31
+second_product,lens_type,none,glasses24,FIXED_AMOUNT,1500,0.0,2020-01-01,2020-02-31
+EOT;
+
+        $filePath = @tempnam(sys_get_temp_dir(), 'cop');
+
+        file_put_contents($filePath, $csvString);
+
+        return $filePath;
+    }
+
+    /**
+     * @return string
+     */
+    private function getInValidCsvFile(): string
```

Comment:
Either abstract away csv reading with a reader that you can mock or use a virtual file system.